### PR TITLE
feat: allow specifying glob option for extra-files

### LINKED ISSUE #1619: Support `extra-files` globs
Sometimes we want to attempt to bulk update N extra files. For example, in https://github.com/chingor13/functions-framework-dotnet/pull/7, we want to update a bunch of `.csproj` files but do not necessarily want to keep updating the config every time we add a new `.csproj` file.

Investigate iterating over files with a file glob:

```json
{
  "extra-files": [
    {
      "type": "xml",
      "glob": "src/**/*.xml",
      "xpath": "//xpath/to/version",
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by chingor13**
When specifying `extra-files` in a manifest, you can add a `glob: true` entry and the `path` specified will be treated as a file glob.

Fixes #1619

### CODE REVIEW

File: src/strategies/base.ts [hunk 1]
```diff
@@ -333,51 +333,96 @@ export abstract class BaseStrategy implements Strategy {
     };
   }
 
-  protected extraFileUpdates(
+  // Helper to convert extra files with globs to the file paths to add
+  private async extraFilePaths(extraFile: ExtraFile): Promise<string[]> {
+    if (typeof extraFile !== 'object') {
```
Comment (tmatsuo) [inline src/strategies/base.ts:338]: Why this happens? Does something pass string to this function?

File: src/strategies/base.ts [hunk 2]
```diff
@@ -333,51 +333,96 @@ export abstract class BaseStrategy implements Strategy {
...
+      return [extraFile];
+    }
+
+    if (!extraFile.glob) {
+      return [extraFile.path];
+    }
+
+    if (extraFile.path.startsWith('/')) {
+      // glob is relative to root, strip the leading `/` for glob matching
```
Comment (tmatsuo) [inline src/strategies/base.ts:347]: I don't understand why the glob doesn't work with the leading slash ...

File: test/fixtures/manifest/config/extra-files-glob.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (tmatsuo) [inline test/fixtures/manifest/config/extra-files-glob.json:1]: Is this file used?

File: src/strategies/base.ts (hunk 1 above)
Comment (chingor13) [reply src/strategies/base.ts:338]: Yes, this config is coming from a config file which is an array of a set of different types -- one of which is a string.

https://github.com/googleapis/release-please/blob/main/schemas/config.json#L114-L178

File: test/fixtures/manifest/config/extra-files-glob.json (hunk 3 above)
Comment (chingor13) [reply test/fixtures/manifest/config/extra-files-glob.json:1]: All config files in that directory are turned into tests that verify that they adhere to the JSONSchema

File: src/strategies/base.ts (hunk 2 above)
Comment (chingor13) [reply src/strategies/base.ts:347]: We could perhaps fix this in the upstream library, but the filenames coming from GitHub do not have a leading `/` (so the leading slash on the glob does not match)

Comment (tmatsuo) [reply src/strategies/base.ts:347]: I see!

File: test/fixtures/manifest/config/extra-files-glob.json (hunk 3 above)
Comment (tmatsuo) [reply test/fixtures/manifest/config/extra-files-glob.json:1]: thx

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**